modules/loader/src/main/java/com/opengamma/strata/loader/csv/BulletPaymentTradeCsvPlugin.java

```diff
@@ -33,22 +34,21 @@
 /**
  * Handles the CSV file format for Bullet Payment trades.
  */
-final class BulletPaymentTradeCsvPlugin implements TradeCsvParserPlugin, TradeTypeCsvWriter<BulletPaymentTrade> {
+final class BulletPaymentTradeCsvPlugin implements TradeCsvParserPlugin, TradeCsvWriterPlugin<BulletPaymentTrade> {
 
   /**
    * The singleton instance of the plugin.
    */
   public static final BulletPaymentTradeCsvPlugin INSTANCE = new BulletPaymentTradeCsvPlugin();
 
   /** The headers. */
-  private static final ImmutableList<String> HEADERS = ImmutableList.<String>builder()
-      .add(DIRECTION_FIELD)
-      .add(CURRENCY_FIELD)
-      .add(NOTIONAL_FIELD)
-      .add(PAYMENT_DATE_FIELD)
-      .add(PAYMENT_DATE_CNV_FIELD)
-      .add(PAYMENT_DATE_CAL_FIELD)
-      .build();
+  private static final LinkedHashSet<String> HEADERS = new LinkedHashSet<>(ImmutableList.of(
+          DIRECTION_FIELD,
+          CURRENCY_FIELD,
+          NOTIONAL_FIELD,
+          PAYMENT_DATE_FIELD,
+          PAYMENT_DATE_CNV_FIELD,
+          PAYMENT_DATE_CAL_FIELD));
```

Comment:
```suggestion
  private static final ImmutableSet<String> HEADERS = ImmutableSet.of(
          DIRECTION_FIELD,
          CURRENCY_FIELD,
          NOTIONAL_FIELD,
          PAYMENT_DATE_FIELD,
          PAYMENT_DATE_CNV_FIELD,
          PAYMENT_DATE_CAL_FIELD);
```

This comment applies to lots of classes.